src/hooks/useDydxClient.tsx

```diff
@@ -112,6 +115,23 @@ const useDydxClientContext = () => {
     })();
   }, [networkConfig]);
 
+  // ------ Gas Denom ------ //
+
+  const [gasDenom, setGasDenom] = useLocalStorage<SelectedGasDenom>({
+    key: LocalStorageKey.SelectedGasDenom,
+    defaultValue: SelectedGasDenom.USDC,
+  });
+
+  const setSelectedGasDenom = useCallback(
```

Comment:
I used `useCallback` because this fn is a dependency in a `useMemo` in `<PreferencesDialog>`. I wanted the `useMemo` to update when `compositeClient` initializes but no further re-renders after.

I could either keep `useCallback` here, or add `compositeClient` as the dep in `<PreferencesDialog>`